### ISSUE TITLE
type juggling is not detecting.

### ISSUE THREAD

**Issue by mukundbhuva** [vimeo/psalm#7841]
Here is the example code : [type juggling](https://psalm.dev/r/87c2dcd1fd)

**Comment by AndrolGenhald** [comment]
Have you tried the [strictBinaryOperands](https://psalm.dev/docs/running_psalm/configuration/#strictbinaryoperands) setting? Unfortunately psalm.dev doesn't support it.

**Comment by lxix** [comment]
Setting `strictBinaryOperands` to true doesn't affects the result for me. Is this even a problem? What issue should be reported?

**Comment by orklah** [comment]
Yeah, strictBinaryOperands is really badly named and doesn't do what everyone suppose it does.

== logic is really not developped much in Psalm (mainly because it's a gigantic mess and edge cases everywhere make the whole thing meaningless except for checking known literals on each side, which is even more meaningless)

I'd recommend using https://github.com/orklah/psalm-strict-equality or something similar to get the most out of Psalm
